src/util/TransitiveModel.hxx

```diff
@@ -82,20 +83,39 @@ public:
     m_Filters[observer].insert(ilist.begin(), ilist.end());
   }
 
+  void prune() final
+  {
+    Subject<Transition<Model>>::prune();
+
+    for (auto it = m_Filters.begin(); it != m_Filters.end(); ++it)
+    {
+      if (std::get<0>(*it).expired())
+      {
+        m_Filters.erase(it);
+      }
+    }
+  }
+
 private:
   using Subject<Notification>::addObserver;
-  using FilterType = Map<ObserverSPtr<Notification>, Set<typename Notification::TransitionType>>;
+  using FilterType = TreeMap<ObserverWPtr<Notification>, Set<typename Notification::TransitionType>,
+                             std::owner_less<ObserverWPtr<Notification>>>;
   FilterType m_Filters;
 
-  bool mustNotify(ObserverSPtr<Notification> &observer, const Notification &event) const noexcept final
+  bool mustNotify(ObserverWPtr<Notification> &observer, const Notification &event) const noexcept final
   {
-    auto it = m_Filters.find(observer);
-    if (it == m_Filters.end())
+    try
     {
-      return true;
+      if (m_Filters.count(observer) > 0)
+      {
+        return m_Filters.at(observer).count(event.getType()) == 1;
+      }
     }
-    return m_Filters.at(observer).count(event.getType()) == 1;
+    catch (...)
+    {
```

Comment:
It is impossible for this code to throw I think, and if it does, it should not be silent

Reply:
I think you are right. I will delete try, catch block.